Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -626,7 +627,14 @@ public override bool TrySpillAt(EntityCoordinates coordinates, Solution solution
             return false;
         }
 
-        var gridUid = coordinates.GetGridUid(EntityManager);
+        var coordsEntity = coordinates.EntityId;
+        if (!TryComp(coordsEntity, out TransformComponent? transform))
+        {
+            puddleUid = EntityUid.Invalid;
+            return false;
+        }
+
+        var gridUid = _transform.GetGrid((coordsEntity, transform));
```

Comment:
```suggestion
        var gridUid = _transform.GetGrid(coordinates);
```
just use the right overload